File: mybatis-plus-sample-id-string/src/main/java/com/baomidou/samples/mapper/UserMapper.java
package com.baomidou.samples.mapper;

import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import com.baomidou.samples.entity.User;
import org.apache.ibatis.annotations.Mapper;

@Mapper
public interface UserMapper extends BaseMapper<User> {

}

File: mybatis-plus-sample-kotlin/src/main/kotlin/com/baomidou/mybatisplus/samples/kotlin/mapper/UserMapper.kt
package com.baomidou.mybatisplus.samples.kotlin.mapper

import com.baomidou.mybatisplus.core.mapper.BaseMapper
import com.baomidou.mybatisplus.samples.kotlin.entity.User
import org.apache.ibatis.annotations.Mapper

/**
 * @author nieqiurong
 */
@Mapper
interface UserMapper : BaseMapper<User> {

    fun user(id: Int): User {
        return selectById(id)
    }

}